### PR TITLE
refactor: plain queue implementation

### PR DESCRIPTION
**Pull request by OjerIsaac**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -6,25 +6,6 @@ This is a bulk email sending service that sends email at a particular time set b
...
-<p align="center">
-  <img src="screenshot.jpg" width="100%" alt="bulk email screenshot" title="image sample">
-</p>
-
-<p align="center">
-  <img src="screenshot2.jpg" width="100%" alt="bulk email screenshot" title="image sample">
-</p>
+## Note
+### Would we need a cron job to get the entire thing working?
+- The code here does not require a cron job to run, as it contains a main loop that continuously processes the email queue until it is empty.
```
Comment (xyluz) [inline README.md:11]: The while loop will only run if the page is open. A queue system should be better made to ensure continuous background work is going on. That's the entire point of queues in this case.

File: queue-system.php [hunk 2]
```diff
@@ -0,0 +1,61 @@
+<?php
+
+require_once 'env.class.php';
+$__DotEnvironment = new DotEnvironment(realpath("./.env"));
```
Comment (xyluz) [inline queue-system.php]: You should have .env.example in the repo

File: queue-system.php [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+
+// Set up a database connection
+$db = new PDO('mysql:host="'.$_ENV['DB_HOST'].'";dbname="'.$_ENV['DB_NAME'].'"', '"'.$_ENV['DB_USER'].'"', '"'.$_ENV['DB_PASSWORD'].'"');
```
Comment (xyluz) [inline queue-system.php]: Move this to it's own class

File: queue-system.php [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+
+// Define a function to add a new email to the queue
+function add_email_to_queue($to, $subject, $message) {
```
Comment (xyluz) [inline queue-system.php]: This does not use proper OOP concepts. The queue should be a class that can be reused anywhere. What you have is a file containing several functions.

File: queue-system.php (hunk 2 above)
Comment (OjerIsaac) [reply queue-system.php]: checked

File: queue-system.php (hunk 3 above)
Comment (OjerIsaac) [reply queue-system.php]: checked

File: queue-system.php (hunk 4 above)
Comment (OjerIsaac) [reply queue-system.php]: checked

File: classes/queue-system.php [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+
+
+    /**
+     * Define a function to add a new email to the queue
+     * @param mixed $to
+     * @param mixed $subject
+     * @param mixed $message
+     * @return void
+     */
+    public function add_email_to_queue($to, $subject, $message) {
```
Comment (xyluz) [inline classes/queue-system.php]: Usually, each method parameter should be "typed" (hence, specify the expected type), and the method's return type should be specified. That way, you won't need to add the comment at the top - the comment at the top of the method should be used to give other information not provided in the method itself.

File: classes/queue-system.php [hunk 6]
```diff
@@ -0,0 +1,103 @@
...
+        return $stmt;
+    }
+
+    /**
+     * Define a function to send email using PHPMailer
+     * 
+     */
+    public function send_email($emailTo, $emailSubject, $emailMessage)
+    {
+        # code...
```
Comment (xyluz) [inline classes/queue-system.php]: Do you intend to do this?

**Review by xyluz — COMMENTED** [review]
This is a good improvement. I think you are ready to join me on the project itself.

File: classes/queue-system.php (hunk 5 above)
Comment (OjerIsaac) [reply classes/queue-system.php]: mmm... i would implement that

File: classes/queue-system.php (hunk 6 above)
Comment (OjerIsaac) [reply classes/queue-system.php]: sure.. would implement that in the next push

File: classes/queue-system.php (hunk 5 above)
Comment (OjerIsaac) [reply classes/queue-system.php]: But php is a dynamically typed language,

**Pull request merged**